### DISCUSSION TITLE
[Request] audiobookshelf

### DISCUSSION THREAD

**Discussion by DSDV** [tteck/Proxmox#1004] [Ideas]
Hello tteck, 

i recently saw TechHut show of audiobookshelf and even thou i often searched for such an application i never found it until he suggested this one. 

Can  you please make a script for it <3 

the project is also on GitHub: https://github.com/advplyr/audiobookshelf

with kind regards

PS: i know you did not provide a paypal.me in the past but i still will offer a donation for this great project :D

**Comment by oOStroudyOo** [comment]
I'd love to see this.

Even though it only takes 2 lines of code to install this app, I just have an obsession with removing all of my applications and installing the Tteck script.

Only thing I guess you would need to add is a switch of some kind to decide if you wanted Nginx installed to. 
ABS advises that it needs to be run through Nginx but I opted to run it through Cloudflare tunnel instead. Also others might already have NPM running else where.

**Reply by DSDV** [reply]
same here - to be hones the only non tteck application right now is my roon server (which i wont request due to it being proprietary) 
i love it so much that there is a unified upgrade path for all my lxc s <2

**Reply by notorioustubze** [reply]
Hi, I’ve Set up the Cloudflare tunnel but when I type the domain into ABS I get error failed to ping server. Any idea what I’m missing? TIA

**Comment by DSDV** [comment]
i guess for LXC the Linux repo installation method would be best https://www.audiobookshelf.org/docs/#linux-install-deb since docker looks like overhead in this case?

**Reply by tteck** [reply]
Will look at

**Comment by tteck** [comment]
Ready for testing
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/audiobookshelf-v5.sh)"
```

**Reply by DSDV** [reply]
Great :D 
ill try it asap <3

**Comment by DSDV** [comment]
![Screenshot from 2023-02-05 19-26-42](https://user-images.githubusercontent.com/5504857/216838687-f3c55e86-f32a-4e9e-81c4-3214b506f940.png)

Work got in the way :D 
finally got around to use your script. 
As allays it worked flawlessly (: 
only had to add cifs-utils afterwards so maybe setting the default to privileged container may be a good idea (:

**Reply by tteck** [reply]
Done!

**Reply by DSDV** [reply]
Also really nice that you add the repolink to the notes section :D 
i was pretty confused to see that it was already there while i wanted to add it ^^

**Comment by mdb17** [comment]
trying to run this script and I get the following
`bash: line 182: exit: A problem occured while trying to create container.: numeric argument required`
not sure if this is a me problem or not.
